vermouth/processors/water_bias.py

```diff
@@ -87,16 +90,20 @@ def assign_residue_water_bias(self, molecule, res_graph):
             _old_resid = res_graph.nodes[res_node]['_old_resid']
             chain = res_graph.nodes[res_node]['chain']
             resname = res_graph.nodes[res_node]['resname']
-
-            if _in_resid_region(_old_resid, self.idr_regions):
-                eps = self.water_bias.get('idr', 0.0)
-                sec_struc = res_graph.nodes[res_node]['cgsecstruct']
+            eps = 0.0
+
+            if len(self.idr_regions)>0:
+                for region in self.idr_regions:
+                    if (region.get('chain') is None or region.get('chain') == chain) and _in_resid_region(_old_resid,
+                                                                                                          region.get(
+                                                                                                                  'resids')):
+                        eps = self.water_bias.get('idr', 0.0)
+                        sec_struc = res_graph.nodes[res_node]['cgsecstruct']
             elif self.auto_bias:
                 sec_struc = res_graph.nodes[res_node]['cgsecstruct']
                 eps = self.water_bias.get(sec_struc, 0.0)
             else:
                 continue
```

Comment:
```suggestion
            if self.auto_bias:
                sec_struc = res_graph.nodes[res_node]['cgsecstruct']
                eps = self.water_bias.get(sec_struc, 0.0)
            else:
                for region in self.idr_regions:
                    if (region.get('chain') is None or region.get('chain') == chain) and _in_resid_region(_old_resid,  region.get('resids')):
                        eps = self.water_bias.get('idr', 0.0)
                        sec_struc = res_graph.nodes[res_node]['cgsecstruct']
```
I think this should do the exact same